test/api/v3/unit/libs/cron.test.js

```diff
@@ -0,0 +1,491 @@
+      expect(user.purchased.plan.consecutive.offset).to.be.empty;
+    });
+  });
+
+  describe('user is sleeping', () => {
+    beforeEach(() => {
+      user.preferences.sleep = true;
+    });
+
+    it('clears user buffs', () => {
+      user.stats.buffs = {
+        str: 1,
+        int: 1,
+        per: 1,
+        con: 1,
+        stealth: 1,
+        streaks: true,
+      };
+
+      cron({user, tasksByType, daysMissed, analytics});
+
+      expect(user.stats.buffs.str).to.equal(0);
+      expect(user.stats.buffs.int).to.equal(0);
+      expect(user.stats.buffs.per).to.equal(0);
+      expect(user.stats.buffs.con).to.equal(0);
+      expect(user.stats.buffs.stealth).to.equal(0);
+      expect(user.stats.buffs.streaks).to.be.false;
+    });
+
+    it('resets all dailies without damaging user', () => {
+      let daily = {
+        text: 'test daily',
+        type: 'daily',
+        frequency: 'daily',
+        everyX: 5,
+        startDate: new Date(),
+      };
+
+      let task = new Tasks.daily(Tasks.Task.sanitize(daily)); // eslint-disable-line babel/new-cap
+      tasksByType.dailys.push(task);
+      tasksByType.dailys[0].completed = true;
+
+      cron({user, tasksByType, daysMissed, analytics});
+
+      expect(tasksByType.dailys[0].completed).to.be.false;
+    });
+  });
+
+  describe('todos', () => {
+    beforeEach(() => {
+      let todo = {
+        text: 'test todo',
+        type: 'todo',
+        value: 0,
+      };
+
+      let task = new Tasks.todo(Tasks.Task.sanitize(todo)); // eslint-disable-line babel/new-cap
+      tasksByType.todos.push(task);
+    });
+
+    it('should make uncompleted todos redder', () => {
+      let valueBefore = tasksByType.todos[0].value;
+      cron({user, tasksByType, daysMissed, analytics});
+      expect(tasksByType.todos[0].value).to.be.lessThan(valueBefore);
+    });
+
+    it('should add history of completed todos to user history', () => {
+      tasksByType.todos[0].completed = true;
+
+      cron({user, tasksByType, daysMissed, analytics});
+
+      expect(user.history.todos).to.be.lengthOf(1);
+    });
+  });
+
+  describe('dailys', () => {
+    beforeEach(() => {
+      let daily = {
+        text: 'test daily',
+        type: 'daily',
+      };
+
+      let task = new Tasks.daily(Tasks.Task.sanitize(daily)); // eslint-disable-line babel/new-cap
+      tasksByType.dailys = [];
+      tasksByType.dailys.push(task);
+
+      user._statsComputed = {
+        con: 1,
+      };
+    });
+
+    it('should add history', () => {
+      cron({user, tasksByType, daysMissed, analytics});
+      expect(tasksByType.dailys[0].history).to.be.lengthOf(1);
+    });
+
+    it('should set tasks completed to false', () => {
+      tasksByType.dailys[0].completed = true;
+      cron({user, tasksByType, daysMissed, analytics});
+      expect(tasksByType.dailys[0].completed).to.be.false;
+    });
+
+    it('should set task checklist to completed for completed dailys', () => {
```

Comment:
Hmm... I see now that you're saying that completed dailies with incomplete checklists should be treated as if all their checklist items were completed. 

I guess the actual saving of the tasks and resetting them to their uncompleted state happens elsewhere?